src/ICSharpCode.SharpZipLib/Zip/Compression/Streams/InflaterInputStream.cs

```diff
@@ -108,6 +108,24 @@ public void SetInflaterInput(Inflater inflater)
 			}
 		}
 
+		/// <summary>
+		/// Resize input buffer according to a specific array size
+		/// </summary>
+		/// <param name="bufferSize"></param>		
+		public void ResizeBuffer(int bufferSize)
+		{
+			if (available == 0)
+			{
+				int oldSize = rawData.Length;
+				byte[] resized = rawData;
+				Array.Resize(ref resized, bufferSize);
```

Comment:
You are aware that is a costly copy operation?

Reply:
Yes, I am. But in this particular case I don't think that it is significant.